dali/pipeline/data/tensor_list.h

```diff
@@ -219,14 +217,14 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
    * the user to manage the lifetime of the allocation such that it
    * persist while it is in use by the Tensor.
    */
-  DLL_PUBLIC inline void ShareData(void *ptr, size_t bytes) {
+  inline void ShareData(const shared_ptr<void> &ptr, size_t bytes, const TensorListShape<> &shape) {
```

Comment:
For completeness - maybe there should also be a type?
```suggestion
  inline void ShareData(const shared_ptr<void> &ptr, size_t bytes, const TensorListShape<> &shape, const TypeInfo &type = {}) {
```

Reply:
...this would be in line with the new Resiz.

Reply:
I wanted to align the API with the Tensor API. It doesn't accept type in the `ShareData` function.

Reply:
Done